### PR TITLE
Use TEfficiency for efficiencies

### PR DESCRIPTION
**Pull request by chiarazampolli**
Needs this one, then I will need to rebase:

https://github.com/AliceO2Group/AliceO2/pull/7293

### CODE REVIEW

**Comment by chiarazampolli** [comment]
Hi,
I just realized that TEfficiency is not visualized by the QCG, so this might not be the right way... 
Chiara

**Comment by chiarazampolli** [comment]
> Hi, I just realized that TEfficiency is not visualized by the QCG, so this might not be the right way... Chiara

Fixed, it matters in which order one adds numerator and denominator.

**Comment by chiarazampolli** [comment]
Ready for review. The PR in QC https://github.com/AliceO2Group/QualityControl/pull/909/files depends then on this one.

**Comment by ktf** [comment]
Hi, it looks like there is a genuine issue in QC.
```
## sw/BUILD/QualityControl-latest/log
--
FAILED: Modules/GLO/CMakeFiles/O2QcGLO.dir/src/ITSTPCMatchingTask.cxx.o 
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/c++ -DBOOST_ALL_NO_LIB -DBOOST_CONTAINER_DYN_LINK -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SERIALIZATION_DYN_LINK -DDPL_ENABLE_BACKTRACE -DFAIRMQ_HAS_STD_FILESYSTEM=1 -DFMT_LOCALE -DO2QcGLO_EXPORTS -D__CLANG_SUPPORT_DYN_ANNOTATION__ -I/sw/SOURCES/QualityControl/v1.34.1/v1.34.1/Modules/GLO/include -I/sw/SOURCES/QualityControl/v1.34.1/v1.34.1/Modules/GLO/src -I/sw/SOURCES/QualityControl/v1.34.1/v1.34.1/Modules/GLO -I/sw/SOURCES/QualityControl/v1.34.1/v1.34.1/Framework/include -isystem /sw/slc8_x86-64/boost/v1.75.0-local1/include -isystem /sw/slc8_x86-64/FairLogger/v1.10.1-local1/include -isystem /sw/slc8_x86-64/fmt/7.1.0-local1/include -isystem /sw/slc8_x86-64/FairMQ/v1.4.42-local2/include/fairmq -isystem /sw/slc8_x86-64/FairMQ/v1.4.42-local2/include -isystem /sw/slc8_x86-64/ROOT/v6-24-06-local1/include -isystem /sw/slc8_x86-64/Common-O2/v1.6.0-local1/include -isystem /sw/slc8_x86-64/libInfoLogger/v2.2.0-local1/include -isystem /sw/slc8_x86-64/Monitoring/v3.8.10-local1/include -isystem /sw/slc8_x86-64/Configuration/v2.6.3-local1/include -isystem /sw/slc8_x86-64/Control-OCCPlugin/v0.27.0-local2/include/Occ -isystem /sw/slc8_x86-64/grpc/v1.34.0-alice2-local1/include -isystem /sw/slc8_x86-64/protobuf/v3.15.8-local1/include -isystem /sw/slc8_x86-64/zlib/v1.2.8-5/include -isystem /sw/slc8_x86-64/c-ares/v1.17.1-local1/include -isystem /sw/slc8_x86-64/re2/2019-09-01-local1/include -isystem /sw/slc8_x86-64/abseil/20200225.2-local1/include -isystem /sw/slc8_x86-64/O2/7321-slc8_x86-64-local1/include -isystem /sw/slc8_x86-64/O2/7321-slc8_x86-64-local1/include/GPU -isystem /sw/slc8_x86-64/FairRoot/v18.4.2-local2/include -isystem /sw/slc8_x86-64/arrow/v5.0.0-alice2-local1/include -isystem /sw/slc8_x86-64/Vc/1.4.1-local1/include -isystem /sw/slc8_x86-64/ms_gsl/3.1.0-local1/include -isystem /sw/slc8_x86-64/libjalienO2/0.1.3-local1/include -isystem /sw/slc8_x86-64/RapidJSON/v1.1.0-alice2-local1/include -isystem /sw/slc8_x86-64/libuv/v1.40.0-local1/include -fPIC -O2 -std=c++17 -Wno-error=deprecated-declarations -Wno-error=unused-function -O2 -g -DNDEBUG -fPIC -std=gnu++17 -MD -MT Modules/GLO/CMakeFiles/O2QcGLO.dir/src/ITSTPCMatchingTask.cxx.o -MF Modules/GLO/CMakeFiles/O2QcGLO.dir/src/ITSTPCMatchingTask.cxx.o.d -o Modules/GLO/CMakeFiles/O2QcGLO.dir/src/ITSTPCMatchingTask.cxx.o -c /sw/SOURCES/QualityControl/v1.34.1/v1.34.1/Modules/GLO/src/ITSTPCMatchingTask.cxx
/sw/SOURCES/QualityControl/v1.34.1/v1.34.1/Modules/GLO/src/ITSTPCMatchingTask.cxx: In member function 'virtual void o2::quality_control_modules::glo::ITSTPCMatchingTask::initialize(o2::framework::InitContext&)':
/sw/SOURCES/QualityControl/v1.34.1/v1.34.1/Modules/GLO/src/ITSTPCMatchingTask.cxx:74:55: error: 'class o2::globaltracking::MatchITSTPCQC' has no member named 'getHistoFractionITSTPCmatch'; did you mean 'getFractionITSTPCmatch'?
   74 |   getObjectsManager()->startPublishing(mMatchITSTPCQC.getHistoFractionITSTPCmatch());
      |                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                       getFractionITSTPCmatch
```

**Comment by chiarazampolli** [comment]
Yes, i might have forgotten to push the fix, i had seen it locally, sorry. Will do it asap.

**Comment by chiarazampolli** [comment]
Hi @ktf ,

The problem is that the CI includes also QC (was it always the case?). I thought that QC needs dev of O2, but not the other way around. 
Anyway the QC PR (which is already opened since a couple of days) states that it needs this PR:

https://github.com/AliceO2Group/QualityControl/pull/909

It is WIP exactly because to compile, this PR in O2 should be merged first.

This means that I cannot fix the error (--> it is fixed in the other PR in QC). The only way would be to approve the two PRs at the same time, or skip/ignore the compilation in QC. 

Chiara

**Comment by ktf** [comment]
Can't you simply keep the old API around and drop it at a later point?

**Comment by chiarazampolli** [comment]
I honestly don’t see the point, it means two PR’s. O2 should bot depend on QC. Should it?

**Comment by ktf** [comment]
QC depends on O2 and IMHO we must guarantee the compatibility remains so for a week / 10 days, since the update pace of FLPs and EPNs are different. Merging QC and O2 together might get the CI green, but it will still fail for those who need to use the old QC.

**Comment by chiarazampolli** [comment]
I don't understand. There is a PR on QC which should be merged once this one it merged. The FLP software and EPN software work with compatible O2 and QC, even if in an independent way. O2 does not depend on QC. I really don't see the point in not merging this one, and have QC tests on O2 (were they always there? Was this a recent change?). Once this is merged, QC will not compile till the other PR is merged. If this is your worry, I can ask @Barthelemy  and @knopers8 to review it already, so that then it will be merged immediately. 
This change here is needed also for the pilot beam. The method that does not work in QC just changed name, so I could in principle keep the old one, but in any case, we need this PR to make the QC work properly - without it, one plot will just be empty.

**Comment by Barthelemy** [comment]
Hi, 

I have replied by email but will also reply here as to inform others who might be interested, because it is actually an interesting discussion :)

# Why test QC in O2 ?

The fact that we test QC in O2 comes from a long discussion and evaluation done in WP3 a few years back. 

The problem is that merging an incompatible, non-backward compatible, change in O2 breaks QC. 

It is not always intended, like here. Most of the time people did not realize that they are breaking their module. We then detect it in another PR or in our own CI but meanwhile people cloned or updated locally and are stuck with a broken QC. More importantly perhaps, we are unable to release the FLP Suite.

After many instances of it, we asked that downstream packages are also tested. And since then it has avoided this type of problem at the cost of having to push backward compatible changes in the rare instances when the change in O2 implies a change in QC.

# Why not just force merge here and quickly merge in QC ? 

This is always possible. It means that until the QC PR has passed the CI, QC is broken for anyone updating/cloning and all other PRs and CI. If we are unlucky, it can be a bit of time. 


I know that it is not 100% satisfactory, but it does not come from nowhere and tries to address in the most efficient way a recurring problem we had.

File: Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h [hunk 1]
```diff
@@ -43,6 +43,7 @@ class MatchITSTPCQC
   void reset();
   TH1F* getHistoPtTPC() const { return mPtTPC; }
   TEfficiency* getFractionITSTPCmatch() const { return mFractionITSTPCmatch; }
+  TEfficiency* getHistoFractionITSTPCmatch() const { return mFractionITSTPCmatch; }
```
Comment (ktf) [inline Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h:46]: Does this actually work? Wasn't the old API using TH1F?

**Comment by chiarazampolli** [comment]
cs8 built error does not seem related to this PR:
```
## sw/BUILD/O2-latest/log
--
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createNewGVNPass()'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::Type::getInt128Ty(llvm::LLVMContext&)'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createTargetTransformInfoWrapperPass(llvm::TargetIRAnalysis)'
collect2: error: ld returned 1 exit status
[822/3372] Linking CXX executable stage/tests/o2-test-framework-Task
FAILED: stage/tests/o2-test-framework-Task 
: && /sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/c++ -fPIC -O2 -std=c++17 -Werror -Wno-error=deprecated-declarations -O2 -g  Framework/Core/CMakeFiles/O2test-framework-Task.dir/test/test_Task.cxx.o -o stage/tests/o2-test-framework-Task -L/sw/slc8_x86-64/FairRoot/v18.4.2-29/lib -Wl,-rpath,/sw/slc8_x86-64/FairRoot/v18.4.2-29/lib:/sw/BUILD/0c43ca1ce3af5abf1826c83c91549581447614e2/O2/stage/lib64:/sw/slc8_x86-64/Common-O2/v1.6.0-19/lib:/sw/slc8_x86-64/Configuration/v2.6.3-7/lib:/sw/slc8_x86-64/Monitoring/v3.8.11-local1/lib:/sw/slc8_x86-64/libInfoLogger/v2.2.0-3/lib:/sw/slc8_x86-64/boost/v1.75.0-19/lib:/sw/slc8_x86-64/ROOT/v6-24-06-11/lib:/sw/slc8_x86-64/libjalienO2/0.1.3-10/lib:/sw/slc8_x86-64/FairMQ/v1.4.42-11/lib:/sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib:/sw/slc8_x86-64/FairLogger/v1.10.1-5/lib:/sw/slc8_x86-64/libuv/v1.40.0-13/lib  stage/lib64/libO2Framework.so  /sw/slc8_x86-64/Common-O2/v1.6.0-19/lib/libCommon.so  /sw/slc8_x86-64/Configuration/v2.6.3-7/lib/libConfiguration.so  /sw/slc8_x86-64/Monitoring/v3.8.11-local1/lib/libO2Monitoring.so  /sw/slc8_x86-64/libInfoLogger/v2.2.0-3/lib/libO2Infologger.so  -lpthread  stage/lib64/libO2CCDB.so  /usr/lib64/libcurl.so  stage/lib64/libO2CommonUtils.so  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_iostreams.so.1.75.0  -lbz2  -llzma  -lz  stage/lib64/libO2CommonDataFormat.so  stage/lib64/libO2MathUtils.so  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGenVector.so.6.24.06  /sw/slc8_x86-64/Vc/1.4.1-13/lib/libVc.a  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libBase.so  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libGeoBase.so  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libVMC.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGeom.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libEG.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libPhysics.so.6.24.06  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libFairTools.so  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libParBase.so  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libParMQ.so  /sw/slc8_x86-64/libjalienO2/0.1.3-10/lib/libjalienO2.so  stage/lib64/libO2FrameworkFoundation.so  stage/lib64/libO2Headers.so  stage/lib64/libO2GPUCommon.so  stage/lib64/libO2MemoryResources.so  /sw/slc8_x86-64/FairMQ/v1.4.42-11/lib/libFairMQ.so.1.4.42  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_filesystem.so.1.75.0  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_program_options.so.1.75.0  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_container.so.1.75.0  -lrt  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_regex.so.1.75.0  /sw/slc8_x86-64/FairMQ/v1.4.42-11/lib/libFairMQStateMachine.so.1.4.42  /sw/slc8_x86-64/FairMQ/v1.4.42-11/lib/libFairMQTools.so.1.4.42  /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libarrow.so.500.0.0  -ldl  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libROOTDataFrame.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libTreePlayer.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libTree.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGraf3d.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGpad.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGraf.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libHist.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libMatrix.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libMathCore.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libROOTVecOps.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libImt.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libMultiProc.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libNet.so.6.24.06  /sw/slc8_x86-64/FairLogger/v1.10.1-5/lib/libFairLogger.so.1.10.1  /sw/slc8_x86-64/fmt/7.1.0-13/lib/libfmt.a  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_serialization.so.1.75.0  /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0  /sw/slc8_x86-64/libuv/v1.40.0-13/lib/libuv.so  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libRIO.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libThread.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libCore.so.6.24.06  -lpthread  -Wl,-rpath-link,/sw/slc8_x86-64/boost/v1.75.0-19/lib && :
--
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createNewGVNPass()'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::Type::getInt128Ty(llvm::LLVMContext&)'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createTargetTransformInfoWrapperPass(llvm::TargetIRAnalysis)'
collect2: error: ld returned 1 exit status
[823/3372] Linking CXX executable stage/tests/o2-test-framework-ExternalFairMQDeviceWorkflow
FAILED: stage/tests/o2-test-framework-ExternalFairMQDeviceWorkflow 
: && /sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/c++ -fPIC -O2 -std=c++17 -Werror -Wno-error=deprecated-declarations -O2 -g  Framework/Core/CMakeFiles/O2test-framework-ExternalFairMQDeviceWorkflow.dir/test/test_ExternalFairMQDeviceWorkflow.cxx.o -o stage/tests/o2-test-framework-ExternalFairMQDeviceWorkflow -L/sw/slc8_x86-64/FairRoot/v18.4.2-29/lib -Wl,-rpath,/sw/slc8_x86-64/FairRoot/v18.4.2-29/lib:/sw/BUILD/0c43ca1ce3af5abf1826c83c91549581447614e2/O2/stage/lib64:/sw/slc8_x86-64/Common-O2/v1.6.0-19/lib:/sw/slc8_x86-64/Configuration/v2.6.3-7/lib:/sw/slc8_x86-64/Monitoring/v3.8.11-local1/lib:/sw/slc8_x86-64/libInfoLogger/v2.2.0-3/lib:/sw/slc8_x86-64/boost/v1.75.0-19/lib:/sw/slc8_x86-64/ROOT/v6-24-06-11/lib:/sw/slc8_x86-64/libjalienO2/0.1.3-10/lib:/sw/slc8_x86-64/FairMQ/v1.4.42-11/lib:/sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib:/sw/slc8_x86-64/FairLogger/v1.10.1-5/lib:/sw/slc8_x86-64/libuv/v1.40.0-13/lib  stage/lib64/libO2Framework.so  /sw/slc8_x86-64/Common-O2/v1.6.0-19/lib/libCommon.so  /sw/slc8_x86-64/Configuration/v2.6.3-7/lib/libConfiguration.so  /sw/slc8_x86-64/Monitoring/v3.8.11-local1/lib/libO2Monitoring.so  /sw/slc8_x86-64/libInfoLogger/v2.2.0-3/lib/libO2Infologger.so  -lpthread  stage/lib64/libO2CCDB.so  /usr/lib64/libcurl.so  stage/lib64/libO2CommonUtils.so  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_iostreams.so.1.75.0  -lbz2  -llzma  -lz  stage/lib64/libO2CommonDataFormat.so  stage/lib64/libO2MathUtils.so  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGenVector.so.6.24.06  /sw/slc8_x86-64/Vc/1.4.1-13/lib/libVc.a  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libBase.so  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libGeoBase.so  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libVMC.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGeom.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libEG.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libPhysics.so.6.24.06  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libFairTools.so  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libParBase.so  /sw/slc8_x86-64/FairRoot/v18.4.2-29/lib/libParMQ.so  /sw/slc8_x86-64/libjalienO2/0.1.3-10/lib/libjalienO2.so  stage/lib64/libO2FrameworkFoundation.so  stage/lib64/libO2Headers.so  stage/lib64/libO2GPUCommon.so  stage/lib64/libO2MemoryResources.so  /sw/slc8_x86-64/FairMQ/v1.4.42-11/lib/libFairMQ.so.1.4.42  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_filesystem.so.1.75.0  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_program_options.so.1.75.0  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_container.so.1.75.0  -lrt  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_regex.so.1.75.0  /sw/slc8_x86-64/FairMQ/v1.4.42-11/lib/libFairMQStateMachine.so.1.4.42  /sw/slc8_x86-64/FairMQ/v1.4.42-11/lib/libFairMQTools.so.1.4.42  /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libarrow.so.500.0.0  -ldl  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libROOTDataFrame.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libTreePlayer.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libTree.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGraf3d.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGpad.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libGraf.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libHist.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libMatrix.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libMathCore.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libROOTVecOps.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libImt.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libMultiProc.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libNet.so.6.24.06  /sw/slc8_x86-64/FairLogger/v1.10.1-5/lib/libFairLogger.so.1.10.1  /sw/slc8_x86-64/fmt/7.1.0-13/lib/libfmt.a  /sw/slc8_x86-64/boost/v1.75.0-19/lib/libboost_serialization.so.1.75.0  /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0  /sw/slc8_x86-64/libuv/v1.40.0-13/lib/libuv.so  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libRIO.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libThread.so.6.24.06  /sw/slc8_x86-64/ROOT/v6-24-06-11/lib/libCore.so.6.24.06  -lpthread  -Wl,-rpath-link,/sw/slc8_x86-64/boost/v1.75.0-19/lib && :
--
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createNewGVNPass()'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::Type::getInt128Ty(llvm::LLVMContext&)'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createTargetTransformInfoWrapperPass(llvm::TargetIRAnalysis)'
collect2: error: ld returned 1 exit status
[824/3372] Building CXX object Detectors/MUON/MCH/TimeClustering/CMakeFiles/O2lib-MCHTimeClustering.dir/src/ROFTimeClusterFinder.cxx.o
[825/3372] Building CXX object Framework/AnalysisSupport/CMakeFiles/O2lib-FrameworkAnalysisSupport.dir/src/AODJAlienReaderHelpers.cxx.o
[826/3372] Linking CXX executable stage/tests/o2-test-framework-SingleDataSource
--
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createNewGVNPass()'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::Type::getInt128Ty(llvm::LLVMContext&)'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createTargetTransformInfoWrapperPass(llvm::TargetIRAnalysis)'
collect2: error: ld returned 1 exit status
[827/3372] Building CXX object Detectors/MUON/MCH/DevIO/Digits/CMakeFiles/MCHDigitIOBaseTask.dir/DigitIOBaseTask.cxx.o
[828/3372] Building CXX object DataFormats/Detectors/MUON/MCH/CMakeFiles/O2test-mch-digit.dir/src/testDigit.cxx.o
[829/3372] Building CXX object Detectors/MUON/MCH/PreClustering/CMakeFiles/O2lib-MCHPreClustering.dir/src/PreClusterFinderMapping.cxx.o
--
[846/3372] Building CXX object Framework/TestWorkflows/CMakeFiles/O2exe-testworkflows-analysis-task-example.dir/src/o2AnalysisTaskExample.cxx.o
[847/3372] Building CXX object Detectors/MUON/MCH/Workflow/CMakeFiles/O2exe-mch-preclusters-sink-workflow.dir/src/PreClusterSinkSpec.cxx.o
[848/3372] Building CXX object Utilities/DataSampling/CMakeFiles/O2exe-datasampling-datasampling-benchmark.dir/test/dataSamplingBenchmark.cxx.o
ninja: build stopped: subcommand failed.
--
```

File: Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h (hunk 1 above)
Comment (chiarazampolli) [reply Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h:46]: The return value is not used, in QC we only do:
```
getObjectsManager()->startPublishing(mMatchITSTPCQC.getFractionITSTPCmatch()); // old was getHistoFractionITSTPCmatch
```

**Comment by chiarazampolli** [comment]
Hello @shahor02 , @ktf , 

The error on o2-cs8 does not look related to this PR, unless I overlooked it (if so, sorry).

Chiara

**Pull request merged**